##### Imports

In [ ]:
import pandas as pd

In [3]:
df = pd.read_csv("../data/raw/membership_groceries_userprofile.csv")

In [4]:
df.columns

Index(['id', 'gender', 'shared_account', 'membership_tier',
       'membership_auto_renew', 'membership_fee', 'membership_start_date',
       'push_notification_enabled', 'have_app', 'app_engagement_score',
       'bought_store_brand', 'promotion_participation_count',
       'average_basket_size', 'use_count', 'reward_points_used'],
      dtype='object')

##### Métricas

In [6]:
df['use_count'].mean()

np.float64(11.301864801864802)

In [10]:
df.groupby('membership_tier')['app_engagement_score'].mean()

membership_tier
premium     52.014759
standard    51.702376
Name: app_engagement_score, dtype: float64

In [11]:
df.groupby('membership_tier')['average_basket_size'].mean()

membership_tier
premium     456.846629
standard    422.657248
Name: average_basket_size, dtype: float64

In [12]:
df['bought_store_brand'].value_counts(normalize=True)

bought_store_brand
0    0.506993
1    0.493007
Name: proportion, dtype: float64

In [13]:
df['reward_points_used'].gt(0).mean()

np.float64(0.9708624708624709)

In [14]:
df['have_app'].mean()

np.float64(0.7482517482517482)

In [15]:
df['push_notification_enabled'].mean()

np.float64(0.5174825174825175)

In [16]:
df.groupby('push_notification_enabled')['app_engagement_score'].mean()

push_notification_enabled
0    33.239758
1    69.165878
Name: app_engagement_score, dtype: float64

In [17]:
df['promotion_participation_count'].mean()

np.float64(2.9696969696969697)

In [18]:
df.groupby('promotion_participation_count')['use_count'].mean()

promotion_participation_count
0    11.244444
1    11.814516
2    10.993902
3    11.089744
4    11.325926
5    11.579439
6    11.729730
7    10.906250
8    10.200000
9    11.333333
Name: use_count, dtype: float64

### Principais Insights

- O uso médio do serviço por cliente é de 11,3 vezes.
- O engajamento com o app é consideravelmente maior para clientes que ativaram notificações push (69,2 vs 33,2).
- Clientes premium apresentam maior engajamento com o app e maior gasto médio por compra em relação aos clientes standard.
- 97% dos clientes utilizam pontos de recompensa, demonstrando forte adoção do programa de benefícios.
- A marca própria é comprada por 49,3% dos clientes, indicando espaço para crescimento da fidelidade à marca interna.
- A média de participação em promoções é de cerca de 3 promoções por cliente, com um leve aumento no uso do serviço conforme aumenta a participação.